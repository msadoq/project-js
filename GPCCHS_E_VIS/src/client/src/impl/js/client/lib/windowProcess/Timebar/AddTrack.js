import classnames from 'classnames';
import React, { Component, PropTypes } from 'react';
import { FormGroup, Form } from 'react-bootstrap';
import styles from './Lefttab.css';
import ColorPicker from '../Editor/Components/ColorPicker';

export default class AddTrack extends Component {

  static propTypes = {
    onClose: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    timelines: PropTypes.array.isRequired,
    sessions: PropTypes.array.isRequired,
    color: PropTypes.string.isRequired,
  }

  state = {
    errorMessage: null,
    color: this.props.color,
  }

  willAddTimeline = (e) => {
    e.preventDefault();
    if (!this.newTimelineIdEl.value) {
      this.setState({ errorMessage: 'Please enter an id for the new track' });
    } else if (this.props.timelines.find(t => t.id === this.newTimelineIdEl.value)) {
      this.setState({ errorMessage: 'This id is already taken' });
    } else {
      this.props.onChange(
        this.newTimelineKindEl.value,
        this.newTimelineIdEl.value,
        this.state.color,
        parseInt(this.newTimelineSessionEl.value, 10)
      );
      if (this.props.onClose) {
        this.props.onClose();
      }
    }
  }

  colorChosen = (c) => {
    this.setState({ color: c });
  }

  render() {
    const { color, sessions } = this.props;
    const { errorMessage } = this.state;

    return (
      <Form
        horizontal
        onSubmit={this.willAddTimeline}
      >
        {
          errorMessage ?
            <p className="text-danger" style={{ fontSize: '1em' }}>{errorMessage}</p>
            : null
        }
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Kind</b>
          <select
            ref={(el) => { this.newTimelineKindEl = el; }}
            className={classnames('form-control', styles.formControl)}
          >
            <option value="session">session</option>
            <option disabled value="dataSet">dataSet</option>
            <option disabled value="recordSet">recordSet</option>
          </select>
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Color</b>
          <ColorPicker color={color} onChange={this.colorChosen} />
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Id</b>
          <input
            ref={(el) => { this.newTimelineIdEl = el; }}
            className={classnames('form-control', styles.formControl)}
            placeholder="Parameter n"
          />
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Session</b>
          <select
            ref={(el) => { this.newTimelineSessionEl = el; }}
            className={classnames('form-control', styles.formControl)}
          >
            { sessions.map((v, i) => <option key={i} value={v.id}>{v.name}</option>) }
          </select>
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>&nbsp;</b>
          <input
            type="submit"
            autoFocus
            value="Add track"
            className={classnames(styles.addTrackButton, 'btn-md', 'btn-primary')}
          />
        </FormGroup>
      </Form>
    );
  }
}
