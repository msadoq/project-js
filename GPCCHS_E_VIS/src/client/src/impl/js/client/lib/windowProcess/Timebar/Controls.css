.controlsUl{
  display: table;
  margin: 0px;
  padding: 0px;
}
.controlsUlRight{
  position: absolute;
  top: 0px;
  right: 15px;
}
.controlsLi{
  display: table-cell;
}
.controlButtonPause{
  font-weight: 600;
}
.controlButtonPlay{
  font-weight: 600;
  font-size: 8px;
  padding-left: 9px;
}
.controlButtonPlay:hover{
  background: #549edd;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
}
.controlButtonActive, .controlButtonActive:hover{
  background: #60a7e5;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
}
.speedPopover{
  width: 200px;
}
.speedInputEl{
  display: inline-block;
  margin-right: 10px;
  width: 100px;
}
