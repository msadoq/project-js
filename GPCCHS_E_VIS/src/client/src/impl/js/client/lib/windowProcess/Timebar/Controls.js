import React, { PureComponent, PropTypes } from 'react';
import ControlsLeft from './ControlsLeft';
import ControlsRight from './ControlsRight';

export default class Controls extends PureComponent {

  static propTypes = {
    isPlaying: PropTypes.bool.isRequired,
    timebarRealTime: PropTypes.bool.isRequired,
    play: PropTypes.func.isRequired,
    pause: PropTypes.func.isRequired,
    updateSpeed: PropTypes.func.isRequired,
    toggleTimesetter: PropTypes.func.isRequired,
    restoreWidth: PropTypes.func.isRequired,
    goNow: PropTypes.func.isRequired,
    jump: PropTypes.func.isRequired,
    messages: PropTypes.array,
    timebarUuid: PropTypes.string.isRequired,
    timebarMode: PropTypes.string.isRequired,
    timebarSpeed: PropTypes.number.isRequired,
    currentSessionExists: PropTypes.bool.isRequired,
    masterTimelineExists: PropTypes.bool.isRequired,
    switchToNormalMode: PropTypes.func.isRequired,
    switchToRealtimeMode: PropTypes.func.isRequired,
    switchToExtensibleMode: PropTypes.func.isRequired,
    switchToFixedMode: PropTypes.func.isRequired,
  }

  render() {
    return (
      <div>
        <ControlsLeft
          isPlaying={this.props.isPlaying}
          play={this.props.play}
          pause={this.props.pause}
          updateSpeed={this.props.updateSpeed}
          toggleTimesetter={this.props.toggleTimesetter}
          restoreWidth={this.props.restoreWidth}
          goNow={this.props.goNow}
          jump={this.props.jump}
          messages={this.props.messages}
          timebarUuid={this.props.timebarUuid}
          timebarSpeed={this.props.timebarSpeed}
        />
        <ControlsRight
          play={this.props.play}
          switchToNormalMode={this.props.switchToNormalMode}
          switchToRealtimeMode={this.props.switchToRealtimeMode}
          switchToExtensibleMode={this.props.switchToExtensibleMode}
          switchToFixedMode={this.props.switchToFixedMode}
          timebarMode={this.props.timebarMode}
          timebarUuid={this.props.timebarUuid}
          timebarRealTime={this.props.timebarRealTime}
          currentSessionExists={this.props.currentSessionExists}
          masterTimelineExists={this.props.masterTimelineExists}
        />
      </div>
    );
  }
}
