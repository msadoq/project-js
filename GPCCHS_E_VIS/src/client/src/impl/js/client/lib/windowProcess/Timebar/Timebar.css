.viewportWrapper{
    padding-top: 28px;
    overflow: hidden
}
.viewportContainer{
    min-height: 100px;
    position: relative;
    background: #fff;
    border: 1px solid #ccc;
    background: #bcd6ff;
    height: 100%;
}
.viewport{
    top: 0px;
    position: absolute;
    background-color: rgba(49, 126, 249,0.3);
    height: 100%;
    cursor: pointer;
}
.viewportDragging{
    background-color: rgba(49, 126, 249,0.5);
    cursor: move;
}
.viewportResizing{
    background-color: rgba(49, 166, 249,0.5);
}
.lower, .upper, .current{
    cursor: col-resize;
    position: absolute;
    height: 100%;
    width: 3px;
}
.current{
  border-left: 2px solid #F45;
}
.lower{
    left: 0px;
    border-left: 1px solid #4c96ff;
}
.upper{
    right: 0px;
    border-right: 1px solid #4c96ff;
}
.lower:before, .upper:before{
    width: 10px;
    height: 10px;
    content: '≡';
    color: #4c96ff;
    transform: rotate(90deg);
    position: absolute;
    top: 48%;
}
.lower:before{
    left: 5px;
}
.upper:before{
    left: -5px;
}

.lowerFormattedTime, .upperFormattedTime, .currentFormattedTime, .timeBeginning, .timeEnd{
    font-size: 0.9em;
    font-weight: bold;
    top: -14px;
    position: absolute;
    user-select: none;
}
.timeBeginning, .timeEnd{
    top: -28px;
    font-size: 0.8em;
    color: rgba(0,0,0,0.6);
}
.timeBeginning{
    left: 4px;
}
.timeEnd{
    right: 4px;
}
.lowerFormattedTime{
    transform: translateX(-50%);
}
.upperFormattedTime{
    right: 0;
    transform: translateX(50%);
}
.currentFormattedTime{
    color: #812;
    white-space: nowrap;
    z-index: 2;
    transform: translateX(-50%);
}
.arrowLeft, .arrowRight{
    z-index: 11;
    position: absolute;
    color: #fff;
    background: #444;
    border-radius: 2px;
    border: none;
    top: 5px;
    width: 20px;
    height: 15px;
    cursor: pointer;
}
.arrowLeft:hover, .arrowRight:hover{
    background: #333;
}
.arrowLeft{
    left: 7px;
}
.arrowLeft:before{
    content: '➜';
    position: absolute;
    transform: scale(-1);
    top: -2px;
    left: 4px;
}
.arrowRight{
    right: 7px;
}
.arrowRight:before{
    content: '➜';
    position: absolute;
    top: -3px;
    right: 4px;
}
.resizeTimebarContainer{
  margin: 0;
  height: 4px;
  border-top: 1px solid #aaa;
  cursor: row-resize;
}
.resizeTimebarContainer:before{
    width: 20px;
    height: 12px;
    line-height: 12px;
    text-align: center;
    content: '≡';
    color: #888;
    position: absolute;
    top: 3px;
    left: 50%;
    font-size: 1.2em;
    background-color: #F1F1F1;
    transform: translateX(-50%);
}
.resizingTimebarContainer:before{
  color: #555;
}
.timelines{
  max-height: 150px;
  overflow-y: auto;
}
.timeline{
  width: 100%;
  margin-bottom: 2px;
  height: 18px;
}
.timelineInfos{
  width: 90%;
  max-width: 200px;
  background-color: rgba(255,255,255,0.8);
  height: 100%;
  margin: 0px auto;
  padding-left: 6px;
}
