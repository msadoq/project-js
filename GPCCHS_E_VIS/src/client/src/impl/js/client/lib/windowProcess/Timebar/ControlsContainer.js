import { connect } from 'react-redux';
import _get from 'lodash/get';
import { getMasterTimelineById } from '../../store/selectors/timebars';
import { getSession } from '../../store/selectors/sessions';
import { getMasterSessionId } from '../../store/selectors/masterSession';
import {
  switchToNormalMode,
  switchToRealtimeMode,
  switchToExtensibleMode,
  switchToFixedMode,
  updateSpeed,
  restoreWidth,
  goNow,
} from '../../store/actions/timebars';

import Controls from './Controls';

export default connect(
  (state, { timebarUuid }) => {
    let currentSessionExists = false;
    let masterTimelineExists = false;
    const masterTimeline = getMasterTimelineById(state, timebarUuid);
    if (masterTimeline) {
      masterTimelineExists = true;
      if (getSession(state, masterTimeline.sessionId)) {
        currentSessionExists = true;
      }
    }

    return {
      messages: _get(state, ['messages', `timeSetter-${timebarUuid}`], null),
      currentSessionExists,
      masterTimelineExists,
      masterTimeline,
      masterSessionId: getMasterSessionId(state),
    };
  },
  {
    switchToNormalMode,
    switchToRealtimeMode,
    switchToExtensibleMode,
    switchToFixedMode,
    updateSpeed,
    restoreWidth,
    goNow,
  }
)(Controls);
