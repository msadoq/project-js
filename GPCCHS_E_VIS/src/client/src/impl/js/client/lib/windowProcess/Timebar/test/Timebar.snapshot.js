import React from 'react';
import renderer from 'react-test-renderer';
import TimelinesFixture from '../../../../test/__mocks__/Timelines';
import TimebarFixture from '../../../../test/__mocks__/Timebar';
import Timebar from '../Timebar';

const propsStub = {
  retrieveFormattedFullDateEl: () => null,
  displayTimesetter: () => null,
  onVerticalScroll: () => null,
  updateCursors: () => null,
  updateViewport: () => null,
  updateMode: () => null,
  pause: () => null,
  play: () => null,
  isPlaying: true,
  visuWindow: TimebarFixture.visuWindow,
  slideWindow: TimebarFixture.slideWindow,
  viewport: {
    lower: 1480417084354,
    upper: 1480417204665,
  },
  timelines: TimelinesFixture,
  timebarSpeed: TimebarFixture.speed,
  timebarMode: TimebarFixture.mode,
  timebarId: TimebarFixture.timebarId,
  verticalScroll: 0,
  widthPx: 900,
};

test('Timebar renders correctly', () => {
  const tree = renderer.create(
    <Timebar {...propsStub} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
