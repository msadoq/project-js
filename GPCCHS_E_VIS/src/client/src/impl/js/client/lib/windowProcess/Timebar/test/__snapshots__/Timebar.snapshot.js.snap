exports[`test Timebar renders correctly 1`] = `
<div
  className="Timebar__viewportWrapper___3gVA7"
  onMouseMove={[Function]}>
  <div
    className="Timebar__viewportContainer___2vJ28"
    onDoubleClick={[Function]}
    onWheel={[Function]}>
    <div
      className="Timebar__visuWindow___LjhS5"
      onMouseDown={[Function]}
      style={
        Object {
          "left": "0.8311791939224177%",
          "width": "16.05090141383581%",
        }
      }>
      <span
        className="Timebar__lower___2ACIR"
        cursor="lower"
        onDoubleClick={[Function]}
        onMouseDown={[Function]} />
      <span
        className="Timebar__handle___3vSP_ Timebar__handleLower___2mdzy Timebar__undisplayed___19V6v"
        cursor="lower"
        onDoubleClick={[Function]}
        onMouseDown={[Function]} />
      <span
        className="Timebar__current___2dnsd"
        cursor="current"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "left": "44.90186940085962%",
          }
        } />
      <span
        className="Timebar__handle___3vSP_ Timebar__handleCurrent___3Me05 Timebar__undisplayed___19V6v"
        cursor="current"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "left": "44.90186940085962%",
          }
        } />
      <span
        className="Timebar__upper___3yNvr"
        cursor="upper"
        onDoubleClick={[Function]}
        onMouseDown={[Function]} />
      <span
        className="Timebar__handle___3vSP_ Timebar__handleUpper___onVIW Timebar__undisplayed___19V6v"
        cursor="upper"
        onDoubleClick={[Function]}
        onMouseDown={[Function]} />
      <span
        className="Timebar__lowerFormattedTime___19g2H Timebar__moved___3JlBL">
        10:58:05.354
      </span>
      <span
        className="Timebar__currentFormattedTime___GmzPy"
        style={
          Object {
            "left": "44.90186940085962%",
          }
        }>
        10:58:14.025
      </span>
      <span
        className="Timebar__upperFormattedTime___1TPb0">
        10:58:24.665
      </span>
    </div>
    <span
      className="Timebar__slide___3CAuZ hidden"
      cursor="slideLower"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "left": "4.009608431481743%",
        }
      }
      title="Ext lower cursor" />
    <span
      className="Timebar__handle___3vSP_ Timebar__handleSlideLower___2aCq6 Timebar__undisplayed___19V6v"
      cursor="slideLower"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "left": "4.009608431481743%",
        }
      } />
    <span
      className="Timebar__slide___3CAuZ hidden"
      cursor="slideUpper"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "left": "20.894182576821738%",
        }
      }
      title="Ext upper cursor" />
    <span
      className="Timebar__handle___3vSP_ Timebar__handleSlideUpper___2OYG8 Timebar__undisplayed___19V6v"
      cursor="slideUpper"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "left": "20.894182576821738%",
        }
      } />
    <div
      className="Timebar__timelines___3m3ZW"
      onScroll={[Function]}>
      <div
        className="Timebar__timeline___3fFrT"
        style={
          Object {
            "background": "#fa1289",
          }
        }>
        <div
          className="Timebar__timelineInfos___2OR3P">
          Session 1
        </div>
      </div>
      <div
        className="Timebar__timeline___3fFrT"
        style={
          Object {
            "background": "#31b0d5",
          }
        }>
        <div
          className="Timebar__timelineInfos___2OR3P">
          Session 2
        </div>
      </div>
    </div>
  </div>
  <div
    className="TimebarScale__timebarRuleContainer___3SHix">
    <div
      className="TimebarScale__timebarRule___Nexmx"
      onMouseDown={[Function]}
      style={
        Object {
          "left": "-36.86612196723492%",
        }
      }>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "0%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:57:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "16.623583878448354%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:57:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "33.24716775689671%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:58:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "49.87075163534506%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:58:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "66.49433551379342%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:58:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "83.11791939224177%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:59:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "99.74150327069012%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:59:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "116.36508714913847%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          10:59:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "132.98867102758683%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:00:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "149.6122549060352%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:00:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "166.23583878448355%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:00:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "182.8594226629319%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:01:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "199.48300654138023%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:01:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "216.1065904198286%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:01:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "232.73017429827695%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:02:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "249.3537581767253%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:02:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "265.97734205517366%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:02:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "282.600925933622%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:03:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "299.2245098120704%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:03:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "315.84809369051874%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:03:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "332.4716775689671%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:04:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "349.09526144741545%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:04:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "365.7188453258638%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:04:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "382.34242920431217%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:05:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "398.96601308276047%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:05:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "415.5895969612088%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:05:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "432.2131808396572%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:06:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "448.83676471810554%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:06:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "465.4603485965539%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:06:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "482.08393247500226%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:07:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "498.7075163534506%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:07:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "515.331100231899%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:07:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "531.9546841103473%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:08:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "548.5782679887957%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:08:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "565.201851867244%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:08:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "581.8254357456924%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:09:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "598.4490196241408%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:09:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "615.0726035025891%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:09:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "631.6961873810375%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:10:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "648.3197712594858%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:10:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "664.9433551379342%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:10:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "681.5669390163825%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:11:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "698.1905228948309%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:11:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "714.8141067732793%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:11:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "731.4376906517276%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:12:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "748.061274530176%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:12:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "764.6848584086243%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:12:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "781.3084422870727%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:13:00
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "797.9320261655209%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:13:20
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
      <div
        className="TimebarScale__scaleBar___2cPp7"
        style={
          Object {
            "left": "814.5556100439693%",
          }
        }>
        <span
          className="TimebarScale__scaleTime___3O7Rr">
          11:13:40
        </span>
        <span
          className="TimebarScale__scaleBar___2cPp7" />
      </div>
    </div>
  </div>
</div>
`;
